Sources/VitaminUIKit/Components/Snackbar/README.md

```diff
@@ -0,0 +1,174 @@
+- a `title` String, that should be short, and is mandatory
+- a `message`String, that should also be as short as possible, and is also mandatory
+- an optional `image` that will be displayed on the left hand sode of the snackbar
+
+If the image is set, the title and message will be placed on the right of the image, and aligned to left
+If not, title and content will be centered
+
+All these properties can be set in constructor, by passing them in a `VitaminSnackbar.BasicConfiguration` struct.
+They can also be set separately afterwards
+
+```swift
+// let's set properties in constructor
+let snackbar = VitaminSnackbar(
+    basicConfiguration: VitaminSnackbar.BasicConfiguration(
+        title: "A short title is best",
+        message: "A message should be short",
+        image: VitaminAssets.Placeholder.landscape.image
+    )
+)
+
+// or let's define an empty VitaminSnackbar and set its properties afterwards
+let snackbar2 = VitaminSnackbar()
+snackbar2.title = "Another short title is best",
+snackbar2.message = "A message should be shorter",
+snackbar2.image = VitaminAssets.Placeholder.landscape.image
+```
+
+### Width of a snackbar
+The width of the snackbar will automattically adjust to content, but the `VitaminSnackbar` has a property to cap the width
+- `maxWidth` Double, whose default value is 320
+
+As for basic properties, it can be set by initializer or afterwards
+
+```swift
+// let's set properties in constructor
+let snackbar = VitaminSnackbar(
+        basicConfiguration: VitaminSnackbar.BasicConfiguration(
+        title: "A short title is best",
+        message: "A message should be short"
+    ),
+    maxWidth: 200.0
+)
+
+// or let's define an empty VitaminSnackbar and set its properties afterwards
+let snackbar2 = VitaminSnackbar()
+snackbar2.title = "Another short title is best",
+snackbar2.message = "A message should be shorter",
+snackbar2.maxWidth = 200.0
+```
+
+
+### Displaying a Snackbar
+If you want to manaully display the snackbar, the `VitaminSnackbar` provides a `display()` function, that will first set opacity of the snackbar (and its subviews) to 0, then fade it in by animating the opacity to 1.
+
+```swift
+// let's set properties in constructor
+let snackbar = VitaminSnackbar(
+    basicConfiguration: VitaminSnackbar.BasicConfiguration(
+        title: "A short title is best",
+        message: "A message should be short",
+        image: VitaminAssets.Placeholder.landscape.image
+    )
+)
+
+// Define here the layout constraints to place your snackbar on the screen
+// for example:
+snackbar.centerXAnchor.constraint(equalTo: self.view.centerXAnchor).isActive = true
+snackbar.bottomAnchor.constraint(
+    equalTo: self.view.bottomAnchor,
+    constant: 10.0
+).isActive = true
+
+// fade you snackbar in
+snackbar.display()
+```
+
+⚠️ This method should only be used before displaaying the snackbar, because if it is already displayed, it will, disappear, then reappear with a fade in effect.
```

Comment:
```suggestion
⚠️ This method should only be used before displaying the snackbar, because if it is already displayed, it will, disappear, then reappear with a fade in effect.
```